ggshield/core/git_shell.py

```diff
@@ -77,7 +86,7 @@ def check_git_dir(wd: Optional[str] = None) -> None:
 
 
 def get_git_root(wd: Optional[str] = None) -> str:
-    return git(["rev-parse", "--show-toplevel"], cwd=wd)
+    return git(["rev-parse", "--git-dir"], cwd=wd)
```

Comment:
This new implementation returns a different path. `--show-toplevel` returns the directory containing the `.git` dir, whereas `--git-dir` returns the `.git` dir itself:

```
~/src/ggshield
$ git rev-parse --show-toplevel 
/home/agateau/src/ggshield

~/src/ggshield
$ git rev-parse --git-dir       
.git

~/src/ggshield
$ cd ggshield/core

~/src/ggshield/ggshield/core
$ git rev-parse --show-toplevel
/home/agateau/src/ggshield

~/src/ggshield/ggshield/core
$ git rev-parse --git-dir      
/home/agateau/src/ggshield/.git
```

I suspect this change breaks [.env loading from `utils.py`](https://github.com/GitGuardian/ggshield/blob/main/ggshield/core/utils.py#L218). But no tests broke, so either I am wrong, or there are some holes in the test coverage for .env loading 😟.

It also does not always return an absolute path, which could be a problem.

Reply:
Good catch. I used this workaround to have a generic solution even when we're not in a working tree, like with pre-receive. Latest version should restore previous behavior.